odoo_tools/tasks/migrate.py

```diff
@@ -0,0 +1,298 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+import collections
+import csv
+import json
+import os
+import sys
+
+import git
+import oca_port
+from invoke import task
+
+from ..utils.path import cd
+from .database import get_db_request_result
+
+ANALYZE_SUBMODULES_TO_SKIP = []
+ANALYZE_SUBMODULES = {
+    # default: {"upstream": "OCA"}
+    "odoo-cloud-platform": {"upstream": "camptocamp"},
+    "odoo-enterprise-addons": {"upstream": "camptocamp"},
+    "wms-workload": {"upstream": "camptocamp"},
+    "odoo-shopinvader": {"upstream": "shopinvader"},
+    "odoo-shopinvader-payment": {"upstream": "shopinvader"},
+    "ddmrp-professional": {"upstream": "ForgeFlow"},
+    "forecasting": {"upstream": "ForgeFlow"},
+}
+
+
+class AnalyzeAddons:
+    """Get a migration report of modules.
+
+    Modules belonging to `odoo/local-src` will be listed as `to migrate`,
+    while OCA modules will be analyzed by `oca-port` PRs to check if they
+    could to be migrated, or if some commits/PRs could be ported.
+    """
+
+    def __init__(self, ctx, from_branch, to_branch, database_name):
+        self.ctx = ctx
```

Comment:
AFAIS this is needed only to use `get_db_request_result` but that requires refactoring of the db tasks.
IMO you should accept a DSN or even an open connection to the DB. That's it.